packages/babel-core/src/transformation/file/options/option-manager.js

```diff
@@ -33,6 +31,48 @@ type MergeOptions = {
   dirname?: string
 };
 
+const optionNames = new Set([
+  "filename",
+  "filenameRelative",
+  "inputSourceMap",
+  "env",
+  "mode",
+  "retainLines",
+  "highlightCode",
+  "suppressDeprecationMessages",
+  "presets",
+  "plugins",
+  "ignore",
+  "only",
+  "code",
+  "metadata",
+  "ast",
+  "extends",
+  "comments",
+  "shouldPrintComment",
+  "wrapPluginVisitorMethod",
+  "compact",
+  "minified",
+  "sourceMaps",
+  "sourceMapTarget",
+  "sourceFileName",
+  "sourceRoot",
+  "babelrc",
+  "sourceType",
+  "auxiliaryCommentBefore",
+  "auxiliaryCommentAfter",
+  "resolveModuleSource",
+  "getModuleId",
+  "moduleRoot",
+  "moduleIds",
+  "moduleId",
+  "passPerPreset",
+  // Deprecate top level parserOpts
```

Comment:
I think I know what this means now (I believe I wrote this by saying we should remove the other options in this set that would only be options for  either the parser or generator.

ex: "comments" vs generatorOpts.comments?